smart_campus/app/views.py

```diff
@@ -1036,3 +1103,128 @@ def travelplan_delete_page(request, pk):
     travelplan.delete()
 
     return HttpResponseRedirect('/travelplans/')
+
+
+@login_required
+def question_list_page(request):
+    if request.user.is_administrator():
+        stations = Station.objects.all()
+        question_list = Question.objects.all()
+    else:
+        stations = Satation.objects.filter(owner_group=request.user.group)
+        question_list = Question.objects.filter(linked_station__in=stations)
+
+    paginator = Paginator(question_list, 10)
+
+    page = request.GET.get('page', 1)
+    try:
+        questions = paginator.page(page)
+    except PageNotAnInteger:
+        questions = paginator.page(1)
+    except EmptyPage:
+        """Page number is out of range"""
+        questions = paginator.page(paginator.num_pages)
```

Comment:
Similar code blocks(line 1119~11126) appear multiple times.
Is it possible to write it as a utility function?

Reply:
Okay, I refactored all the code block related this part.